src/H5.c

```diff
@@ -1262,7 +1219,10 @@ H5resize_memory(void *mem, size_t size)
     FUNC_ENTER_API_NOINIT
     H5TRACE2("*x", "*xz", mem, size);
 
-    ret_value = H5MM_realloc(mem, size);
+    if (size != 0)
+        ret_value = H5MM_realloc(mem, size);
+    else if (mem)
+        ret_value = H5MM_xfree(mem);
```

Comment:
I read here https://www.tutorialspoint.com/c_standard_library/c_function_realloc.htm, it said
size − This is the new size for the memory block, in bytes. If it is 0 and ptr points to an existing block of memory, the memory block pointed by ptr is deallocated and a NULL pointer is returned.
Do we need to use H5MM_xfree instead?